src/CrawlerProcessor/JobProcessor.php

```diff
@@ -58,26 +59,55 @@ class JobProcessor implements ProcessorInterface
      * @var \SlmQueue\Controller\Plugin\QueuePlugin
      */
     private $queuePlugin;
-    
+
+    /**
+     * @var string
+     */
+    private $lockDir;
+
+    /**
+     * @var Filesystem
+     */
+    private $fs;
+
+    /**
+     * @var string
+     */
+    private $currentLockFile;
+
     /**
      * @param JsonFetch $jsonFetch
      * @param PlainTextFetch $plainTextFetch
      * @param JobRepository $jobRepository
      * @param HydrationInterface $jobHydrator
      * @param InputFilterInterface $dataInputFilter
+     * @param Filesystem $filesystem
+     * @param string $lockDir
      */
     public function __construct(
         JsonFetch $jsonFetch,
         PlainTextFetch $plainTextFetch,
         JobRepository $jobRepository,
         HydrationInterface $jobHydrator,
-        InputFilterInterface $dataInputFilter)
+        InputFilterInterface $dataInputFilter,
+        Filesystem $filesystem,
+        $lockDir
+    )
     {
         $this->jsonFetch = $jsonFetch;
         $this->plainTextFetch = $plainTextFetch;
         $this->jobRepository = $jobRepository;
         $this->jobHydrator = $jobHydrator;
         $this->dataInputFilter = $dataInputFilter;
+        $this->fs = $filesystem;
+
+        if(!is_dir($lockDir) || !is_writable($lockDir)){
+            throw new \InvalidArgumentException(sprintf(
+                'Simple import lock dir "%s" is not exists or writable.',
```

Comment:
Although this is not crucial, it would be nice, if this was not phrased... strange.

Maybe "does not exists or is not writeable" ?